ci/nvim-lspconfig/default.nix

```
@@ -1,42 +1,95 @@
 {
-  lib,
   callPackage,
   vimPlugins,
   neovimUtils,
   wrapNeovimUnstable,
   neovim-unwrapped,
   runCommand,
   pandoc,
-  python3,
+  jq,
 }:
 let
   nvimConfig = neovimUtils.makeNeovimConfig {
     plugins = [
-      {
-        plugin = vimPlugins.nvim-lspconfig;
-        config = null;
-        optional = false;
-      }
+      # HACK: symlink lspconfig's lsp directory into a "normal" lua plugin
+      # This allows us to `require('wrapped-lsp.foo')`
+      (runCommand "wrapped-lspconfig" { } ''
+        mkdir -p $out/lua
+        ln -s ${vimPlugins.nvim-lspconfig}/lsp $out/lua/wrapped-lsp
+      '')
+      # Some server configs also need lspconfig itself
+      vimPlugins.nvim-lspconfig
     ];
   };
 
   nvim = (wrapNeovimUnstable neovim-unwrapped nvimConfig).overrideAttrs {
     dontFixup = true;
   };
-
 in
+
 runCommand "lspconfig-servers"
   {
-    lspconfig = "${vimPlugins.nvim-lspconfig}";
+    lspconfig = vimPlugins.nvim-lspconfig;
     nativeBuildInputs = [
+      jq
+      nvim
       pandoc
-      python3
     ];
     passthru.unsupported = callPackage ./unsupported.nix { };
   }
   ''
-    export HOME=$(realpath .)
-    # Generates `lsp.json`
-    ${lib.getExe nvim} -u NONE -E -R --headless +'luafile ${./lspconfig-servers.lua}' +q
-    LUA_FILTER=${./desc-filter.lua} python3 ${./clean-desc.py} "lsp.json" >$out
+    for file in "$lspconfig"/lsp/*.lua
+    do
+      name=$(basename --suffix=.lua "$file")
+
+      # A lua @brief doc-comment has the description
+      description=$(
+        awk '
+          # Capture the @brief doc-comment
+          /^---@brief/ {
+            inbrief=1
+            next
+          }
+
+          # Print each line in the doc-comment
+          inbrief && /^--- / {
+            sub(/^--- /, "")
+            print
+            next
+          }
+
+          # Until the end of the comment
+          inbrief && !/^---/ {
+            inbrief=0
+          }
+        ' "$file" \
+        | pandoc -t markdown --lua-filter ${./desc-filter.lua}
+      )
+
+      # Extract actual lua values from the server config
+      (
+        export HOME=$(mktemp -d)
+        info=$(mktemp)
+        err=$(mktemp)
+
+        nvim -u NONE -R --headless -l ${./server-config.lua} "$name" \
+          1> "$info" 2> "$err"
+
+        # If we captured any stderr, it's probably a deprecation notice from lspconfig
+        if [ -s "$err" ]; then
+          jq --rawfile err "$err" '.errors = $err' < "$info"
+        else
+          cat "$info"
+        fi
+
+        rm -rf "$HOME" "$info" "$err"
+      ) |
+
+        # Map each server config to {name: {description, cmd, errors}}
```

Is it still `description`, `cmd` and `errors`? It seems to be just `description` here. (I'm not familiar with `jq` so please pardon my ignorance).

---

`command` and `errors` are piped in from the above subshell.

The jq query `. | .description = $desc` is combining `.` (from stdin) and adding an additional key `"description"`.

However, you have spotted a typo:

```suggestion
        # Map each server config to {name: {description, command, errors}}
```

---

I would normally format the pipe as

```sh
) | jq \
```

or 

```sh
) | \
  jq ...
```

However, interspersing comments breaks that and bash treats them as an EOL